pkg/server/auth/auth.go

```diff
@@ -188,14 +188,17 @@ func WithAPIAuth(next http.Handler, srv *AuthServer, publicRoutes []string) http
 			}
 
 		case UserAccount:
-			if featureflags.Get(FeatureFlagClusterUser) == FeatureFlagSet {
+			if featureflags.IsSet(FeatureFlagClusterUser) {
 				adminAuth := NewJWTAdminCookiePrincipalGetter(srv.Log, srv.tokenSignerVerifier, IDTokenCookieName)
 				multi.Getters = append(multi.Getters, adminAuth)
 			}
 
 		case TokenPassthrough:
 			tokenAuth := NewBearerTokenPassthroughPrincipalGetter(srv.Log, nil, AuthorizationTokenHeaderName, srv.kubernetesClient)
 			multi.Getters = append(multi.Getters, tokenAuth)
+
+		case Anonymous:
+			multi.Getters = []PrincipalGetter{NewAnonymousPrincipalGetter(srv.Log, srv.noAuthUser)}
```

Comment:
Should this be wrapped in a `if featureflags.IsSet(FeatureFlagAnon) {` kind of thing?

Reply:
Wrapped it, we should never get there and this code could use some refactoring..